backend/Api/Staffing/ConsultantController.cs

```diff
@@ -39,69 +40,51 @@ public ConsultantController(ApplicationContext context, IMemoryCache cache)
         var readModels = new ReadModelFactory(service).GetConsultantReadModelsForWeeks(orgUrlKey, weekSet);
         return Ok(readModels);
     }
-    
+
     [HttpPut]
-    [Route("staffing/{staffingId}")]
-    public ActionResult<List<ConsultantReadModel>>Put(
-            [FromRoute] string orgUrlKey,
-            [FromRoute] int staffingId,
-            [FromQuery(Name = "Type")] BookingType bookingType,
-            [FromQuery(Name = "Hours")] double hours = 0
-        )
+    [Route("staffing/update")]
+    public ActionResult<List<ConsultantReadModel>> Put(
+        [FromRoute] string orgUrlKey,
+        [FromBody] StaffingWriteModel staffingWriteModel
+    )
     {
         var service = new StorageService(_cache, _context);
 
-        switch (bookingType)
+        if (!StaffingControllerValidator.validateStaffingWriteModel(staffingWriteModel, service, orgUrlKey))
         {
-            case BookingType.Booking:
-            case BookingType.Offer:
-                service.UpdateStaffing(staffingId, hours);
-                break;
-            case BookingType.PlannedAbsence:
-                service.UpdateAbsence(staffingId, hours);
-                break;
-            case BookingType.Vacation:
-                break;
-            default:
-                throw new ArgumentOutOfRangeException(nameof(bookingType), bookingType, "Invalid bookingType");
+            return BadRequest();
         }
 
-        return Ok(hours);
-    }
-    
-    [HttpPost]
-    [Route("staffing/new")]
-    public ActionResult<List<ConsultantReadModel>>Post(
-        [FromRoute] string orgUrlKey,
-        [FromQuery(Name = "Type")] BookingType bookingType,
-        [FromQuery(Name = "ConsultantID")] int consultantId,
-        [FromQuery(Name = "EngagementID")] int engagementId,
-        [FromQuery(Name = "Year")] int selectedYearParam,
-        [FromQuery(Name = "Week")] int selectedWeekParam,
-        [FromQuery(Name = "Hours")] double hours = 0
-    )
-    {
-        var service = new StorageService(_cache, _context);
+        try
+        {
+            var selectedWeek = new Week(staffingWriteModel.Year, staffingWriteModel.Week);
 
-        var newId = 0;
-        
-        var selectedWeek = new Week((int)selectedYearParam, (int)selectedWeekParam);
+            switch (staffingWriteModel.Type)
+            {
+                case BookingType.Booking:
+                case BookingType.Offer:
+                    service.UpdateOrCreateStaffing(new StaffingKey(staffingWriteModel.EngagementId, staffingWriteModel.ConsultantId, selectedWeek), staffingWriteModel.Hours, orgUrlKey);
+                    break;
```

Comment:
Funker dette for både Booking og Offer? Stilig syntaks isåfall :)

Reply:
Har prøvd begge og begge funka, så ser sånn ut! 😄

Reply:
Kult!